patches/v5/3-remove-static-query-component.patch

```diff
@@ -0,0 +1,105 @@
+diff --git a/packages/gatsby/cache-dir/gatsby-browser-entry.js b/packages/gatsby/cache-dir/gatsby-browser-entry.js
+index 3cce70589e..ee403f129c 100644
+--- a/packages/gatsby/cache-dir/gatsby-browser-entry.js
++++ b/packages/gatsby/cache-dir/gatsby-browser-entry.js
+@@ -1,41 +1,10 @@
+ import React from "react"
+-import PropTypes from "prop-types"
+ import loader from "./loader"
+ 
+ const prefetchPathname = loader.enqueue
+ 
+ const StaticQueryContext = React.createContext({})
+ 
+-function StaticQueryDataRenderer({ staticQueryData, data, query, render }) {
```

Comment:
This patch will become problematic for us to keep in sync with partial hydration and slices.

I would check if you can wrap if(gatsby major) around the export. If not I would refactor this code to a separate file and re-export it inside gatsby-browser-entry and create a patch to delete only that line.

Reply:
> I would check if you can wrap if(gatsby major) around the export

Before doing the patch I thought about doing this, too, but from my Google research I found that this would probably not work. So I'll go for your second option 👍

Reply:
in native esm it doesn't but don't forget we bundle with webpack which can remove code before it's ran

Reply:
@wardpeet done